pidgin/app.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+import flask
+import requests
+
+
+app = flask.Flask(__name__)
+
+
+# Endpoint to get core metadata as JSON from an object_id.
+@app.route('/json/<path:object_id>') # 'path' allows the use of '/' in the id
+def get_json_metadata(object_id):
+    try:
+        metadata = get_metadata_dict(object_id)
+        return translate_dict_to_json(metadata)
+    except Exception as e:
+        return str(e), 300
+
+# Endpoint to get core metadata as BibTeX from an object_id.
+@app.route('/bibtex/<path:object_id>') # 'path' allows the use of '/' in the id
+def get_bibtex_metadata(object_id):
+    metadata = get_metadata_dict(object_id)
+    return translate_dict_to_bibtex(metadata)
+
+
+# Create a dictionary containing the metadata for a given object_id.
+def get_metadata_dict(object_id):
+    response = request_metadata(object_id) # query to peregrine
+    data = response.get_json()
+    return flatten_dict(data) # translate the response to a dictionary
+
+
+# Translate a dictionary to a JSON string.
+def translate_dict_to_json(d):
+    json_str = json.dumps(d)
+    return json_str
+
+
+# Translate a dictionary to a BibTeX string.
+def translate_dict_to_bibtex(d):
+    bibtex_str = '@misc {' + d['object_id'] + ', '
+    for k, v in d.items(): # add each pair to the BibTeX output
+        bibtex_str += k + ' = "' + str(v) + '", '
```

Comment:
This is kind of hard to read; I'd suggest `format`:
```python
    bibtex_str += '{key} = "{value}", '.format(key=k, value=v)
```
or even you could just use a comprehension, and the `string.join` method:
```python
items = ['{} = "{}"'.format(k, v) for k, v in d.items()]
bibtex_items = ', '.join(items)
bibtex_str = '@misc {' + bibtex_items + '}'
```
(you might want to check that since this is just off the top of my head, but hopefully that gives you the idea)